package.json

```diff
@@ -18,6 +18,10 @@
   },
   "sideEffects": false,
   "main": "dist/index.js",
+  "exports": {
+    "require": "dist/index.js",
```

Comment:
require will trigger warnings on old node 13 versions. Looks like there is a solution around this
https://twitter.com/lukastaegert/status/1236188083621687298

Reply:
Ah, here's what rollup end up with. Probably no need in mjs wrapper which would not work when exports support will come to bundlers.
https://github.com/rollup/rollup/pull/3430/files

Reply:
Thanks for the hint. Although I’m definitely not planning to add workarounds for experimental APIs of old non-LTS node versions.

Reply:
Also: how does the rollup solution prevent the dual package hazard?

Reply:
I'm not sure. Looks like some fields are just ignored in early node 13.
https://twitter.com/guybedford/status/1235329049779634177
Or what do you mean by dual package hazard?

Reply:
https://nodejs.org/docs/latest/api/esm.html#esm_dual_package_hazard

And if you want to dig deeper: https://github.com/nodejs/modules/issues/371

Reply:
uuid is stateless package. A few instances should not be a problem. It's the same as a few versions.

Reply:
> uuid is stateless package.

** cough ** [stateless](https://github.com/uuidjs/uuid/blob/master/src/v1.js#L9-L14) ** cough **

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2F80TEu4wOBdPLG%2Fgiphy.gif&token=oops" data-gitmeme-token="oops"><img src="https://media0.giphy.com/media/80TEu4wOBdPLG/giphy.gif" title="Created by gitme.me with /oops"/></a>

Reply:
is it not?

Reply:
Well... it doesn't expose state externally, but v1 has those private state vars it maintains. For functional programming purists and anyone looking for idempotent behavior, v1 isn't gonna cut it.

Reply:
got it. thanks.
well, its on dependent packages to provide es modules support. we can only ask to open issues on those packages instead.

Reply:
I think there are 2 aspects w.r.t. uuid and the dual package hazard:

1. I agree that most likely in practice it wouldn't be a problem to just not care about the `v1()` internal state…
1. On the other hand, `uuid` is so heavily depended upon that it is probably one of the very few modules on npm where the dual package hazard will actually manifest in the real world! I just checked a couple of normal web projects and found that `uuid` often appears up to 10-20 times as a transitive dependency: So unless the entire dependency chain was either 100% ESModule or 100% CommonJS (and that will likely not be the case for a very long time) the dual package hazard will manifest for real!

Again, I think it's unlikely that this will cause bugs in the real world, but it still somewhat hurts the pedantic and principled software engineering regions of my brains 😂

Reply:
Funny thing is that uuid in transitive dependencies may be v3 for ages which is not much different from dual packages hazard and definitely not our responsibility. It's also possible the most used api is stateless v4.

Reply:
That’s a very valid point and in practice probably much more important than my musings about the dual package hazard, I totally agree with you. 

But that still doesn’t convince me that we shouldn’t care about it and at least be exactly aware of what we are doing 😉